.github/workflows/ci.yml

```diff
@@ -14,35 +14,60 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 jobs:
-  build:    
+  prepare:
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/build
+        with:
+          project-id: ${{ secrets.PROJECT_ID }}
+
+  test:    
+    needs: prepare
     runs-on: macos-12
     strategy:
+      fail-fast: false
```

Comment:
Without this, if one job in the matrix strategy fails, it will cancel the rest of the builds. I thought it might be valuable to get back results from all the tests job.